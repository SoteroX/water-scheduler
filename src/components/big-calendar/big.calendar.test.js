import React from "react";
import { shallow } from "enzyme";
// Import mocks
import {
  mockEmptyEvents,
  mockEvents,
  mockSortedPlants
} from "../../__mocks__/mocks";
// Import components
import BigCalendarContainer from "./big.calendar.container";
import BigCalendar from "./big.calendar.component";

describe("App component", () => {
  let wrapper;
  let bigCalendarCrash;
  const mockSetState = jest.fn();

  jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useState: value => [true, mockSetState]
  }));

  beforeEach(() => {
    bigCalendarCrash = <BigCalendar events={mockEmptyEvents} />;

    wrapper = shallow(<BigCalendarContainer sortedPlants={mockSortedPlants} />);
    console.log("wrapper:", wrapper);
  });

  // Makes sure BigCalendarContainer component is render and BigCalendar component crashes
  it("renders BigCalendarContainer component but big calendar component crashes", () => {
    expect(wrapper.contains(bigCalendarCrash)).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });

  // Makes sure BigCalendarContainer component is render
  it("renders BigCalendarContainer component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
