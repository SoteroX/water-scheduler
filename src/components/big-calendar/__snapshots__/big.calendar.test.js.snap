// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App component but big calendar component crashes 1`] = `
<div>
  <div>
    Calendar component couldn't be render
  </div>
</div>
`;

exports[`App component renders BigCalendarContainer component but big calendar component crashes 1`] = `
<div>
  <div>
    Calendar component couldn't be render
  </div>
</div>
`;

exports[`App component renders BigCalendarContainer component without crashing 1`] = `
<div>
  <div>
    Calendar component couldn't be render
  </div>
</div>
`;
